qcelemental/models/molecule.py

```diff
@@ -74,44 +74,172 @@ class Config(ProtoModel.Config):
+                    "a Fractal Server or previously serialized Molecules."
+    )
 
     # Required data
-    symbols: Array[str]
-    geometry: Array[float]
+    symbols: Array[str] = Schema(
+        ...,
+        # LNN Note: "Ordered" might not be the least ambiguous term
+        # (e.g. "ordered" as in linked ordering or "ordered" as in alphabetical?)
+        description="An ordered 1-D array-like object of atomic elemental symbols of shape (N). Index of this list "
+                    "sets atomic order for all other per-atom setting like ``real`` and the first dimension of "
+                    "``geometry``. Ghost/virtual atoms "
+                    "must have an entry in this array-like and are indicated by the matching index in ``real``"
+                    "field."
+    )
+    geometry: Array[float] = Schema(
+        ...,
+        description="An ordered 2-D array-like object of shape (N,3) for atomic coordinates. Index of the "
+                    "first dimension matches the index of all other per-atom settings like ``symbols`` and ``real``."
+    )
 
     # Molecule data
-    name: str = ""
-    identifiers: Optional[Identifiers] = None
-    comment: Optional[str] = None
-    molecular_charge: float = 0.0
-    molecular_multiplicity: int = 1
+    name: str = Schema(
+        "",
+        description="A common or human-readable name to assign to this molecule. Can be arbitrary."
+    )
+    identifiers: Optional[Identifiers] = Schema(
+        None,
+        description="An optional dictionary of additional identifiers which this Molecule can be referenced by "
+                    "such as INCHI, canonical SMILES, etc. See the :class:``Identifiers`` model for more details."
+    )
+    comment: Optional[str] = Schema(
+        None,
+        description="Additional comments for this Molecule. Intended for pure human/user consumption "
+                    "and clarity."
+    )
+    molecular_charge: float = Schema(
+        0.0,
+        description="The net electrostatic charge of this Molecule."
+    )
+    molecular_multiplicity: int = Schema(
+        1,
+        description="The total multiplicity of this Molecule."
+    )
 
     # Atom data
-    masses: Optional[Array[float]] = None
-    real: Optional[Array[bool]] = None
-    atom_labels: Optional[Array[str]] = None
-    atomic_numbers: Optional[Array[np.int16]] = None
-    mass_numbers: Optional[Array[np.int16]] = None
+    masses: Optional[Array[float]] = Schema(
+        None,
+        description="An ordered 1-D array-like object of atomic masses of shape (N). Index "
+                    "matches the index of all other per-atom settings like ``symbols`` and ``real``. If this is not "
+                    "provided, the mass of each atom is inferred from their most common isotope. If this is provided,"
+                    "every atom must have its mass specified, even if they are the common isotopes."
```

Comment:
* missing `[u]`
* "not provided, the mass of the most abundant isotope is used."
* the validator can handle `[None, 2.014, None]` afaik
* "A 1-D array-like object of atomic masses [u] of shape (nat). Atom order in array-like matches other per-atom fields like `symbols` and `geometry`."

Reply:
> the validator can handle [None, 2.014, None] afaik

I don't see anywhere that it would not accept that, but I also don't see anywhere which would convert the `None` inside the list to the correct mass. The only auto-mass assignment I can see are these two lines:

```python
        if values["masses"] is None:  # Setup masses before fixing the orientation
            values["masses"] = np.array([periodictable.to_mass(x) for x in values["symbols"]])
```

which would only trigger if the `masses` field is None as a whole, not a list which has None entries.

Reply:
```
>>> import numpy as np
>>> np.array([4, 3]) is None
False
```

Reply:
Yes, but I think the point was that if you pass in something like 

```
symbols=["C", "C", "C"]
masses=[None, 12.011, None]
```

at which point are the `None` in the List converted to the correct mass based on symbol? The only checks I can see are:

* Is the masses object None?
* Does the masses List shape equal that of the symbols list?

Reply:
the `to_schema(from_schema())` should pass most everything that doesn't have validation turned off through `molparse.from_arrays()` which can handle the `[None, 12.011, None]` case. haven't explicitly tried it, though.

Reply:
```
>>> [None, 12.011, None] is None
False
```

None is not checking the elements, it only checking the *object*.

Reply:
> None is not checking the elements, it only checking the object.

Thats not what I have concerns about, in the original point Lori made

> the validator can handle [None, 2.014, None] afaik

Which I agree with, the validator *can* handle that and will accept that as a valid input, so long as the length of masses matches the length of symbols. My concern though however is that those `None` entries in the list *are accepted/not converted in the first place*, resulting in the masses having undefined/None entries on some of the atoms.

I think there will be a problem, for instance, if we are orienting the molecule and there are `None` in the masses list, when we get to the line `new_geometry -= np.average(new_geometry, axis=0, weights=np_mass)`, then it will throw a type error.

Reply:
Thinking about it some more, the type coercion on `Optional[Array[float]]` won't allow things like `[None 1.2345, None]`, so this is likely a moot point and the validator will reject such energies.